src/swarm-tui/Swarm/TUI/View/Robot.hs

```diff
@@ -50,86 +45,139 @@ import Swarm.Game.State.Substate
 import Swarm.Game.Tick (addTicks)
 import Swarm.Game.Universe
 import Swarm.Game.World.Coords
-import Swarm.TUI.Model.DebugOption
 import Swarm.TUI.Model.Name
 import Swarm.TUI.Model.UI.Gameplay
 import Swarm.TUI.View.Attribute.Attr
 import Swarm.TUI.View.CellDisplay
 import Swarm.TUI.View.Robot.Details
 import Swarm.TUI.View.Robot.Type
 import Swarm.TUI.View.Shared (tabControlFooter)
-import Swarm.Util (applyWhen, maximum0, maximumNE)
+import Swarm.Util (applyWhen, maximum0)
 import Swarm.Util.UnitInterval
 import Swarm.Util.WindowedCounter qualified as WC
 import System.Clock (TimeSpec (..))
 
-extractColWidth :: ColWidth -> Int
-extractColWidth (ColW x) = x
-
-getMaxWidth :: ColWidth -> ColWidth -> ColWidth
-getMaxWidth (ColW w1) (ColW w2) = ColW $ max w1 w2
-
-data RobotRenderingContext = RobotRenderingContext
-  { _mygs :: GameState
-  , _gameplay :: UIGameplay
-  , _timing :: UITiming
-  , _uiDbg :: Set DebugOption
-  }
-
-makeLenses ''RobotRenderingContext
-
-mkRobotDisplay :: RobotRenderingContext -> MixedTabularList Name RobotWidgetRow Widths
-mkRobotDisplay c =
-  mixedTabularList
-    (RobotsListDialog RobotList)
-    (mkLibraryEntries c)
-    (LstItmH 1)
-    (computeColumnWidths $ c ^. uiDbg)
-    assignRowWidthForConstructor
-
-emptyRobotDisplay :: Set DebugOption -> RobotListContent
-emptyRobotDisplay uiDebug =
-  RobotListContent
-    { _robotsListWidget = mixedTabularList (RobotsListDialog RobotList) mempty (LstItmH 1) (computeColumnWidths uiDebug) assignRowWidthForConstructor
-    , _robotsListRenderers =
-        MixedRenderers
-          { cell = drawCell uiDebug
-          , rowHdr = Just rowHdr
-          , colHdr = Just $ colHdr uiDebug
-          , colHdrRowHdr = Just $ ColHdrRowHdr $ \_ _ -> vLimit 1 (fill ' ') <=> hBorder
-          }
+{--------------------------------------------------------------------
+NEW GRID LIST
+--------------------------------------------------------------------}
+
+data RobotColumn
+  = ColID
+  | ColName
+  | ColAge
+  | ColPos
+  | ColItems
+  | ColStatus
+  | ColActns
+  | ColCmds
+  | ColCycles
+  | ColActivity
+  | ColLog
+  deriving (Eq, Ord, Enum, Bounded, Show)
+
+colName :: RobotColumn -> Text
+colName = T.pack . dropPrefix "Col" . show
+
+colWidth :: RobotColumn -> Int
+colWidth = textWidth . colName
+
+-- | TODO: optionally hide ID based on Debug Options
+colWidths :: Seq ColWidth
```

Comment:
I am considering moving the optional RID to the end, so that we only have to check debug options once.